app/models/medium.rb

```diff
@@ -12,6 +12,7 @@ class MissingPutFilePath < StandardError; end
   before_destroy :queue_medium_removal, unless: :external_link
 
   ALLOWED_EXPORT_CONTENT_TYPES = %w(text/csv).freeze
+  ALLOWED_CONTENT_TYPES = /jp(e)?g|gif|png|ico|txt|mp(3|4)|webm|og(a|g|m|v|x)|spx|opus|pdf|ttf|tar|gz|tgz|bz2|tbz2|zip|csv|mp(e)?g|svg|plain|m4(a|b)|json/
```

Comment:
Did you consider inverting this check to a block list of non-allowed content-types vs an inclusive allow list? 

Allow lists are harder to maintain as they are more restrictive but can provide finer controls. 

If we do the allow list I think this regex should match what we are testing, i.e the list in and no more.  I would remove the spx, opus, ttf, tar, gz, bz2, tbz2, zip etc. Also i would ensure we test the content types in full not substrings (can smuggle by faking content types, e.g. `test/html-json` will pass).

Further we could switch from a regex to an dict (O(1) lookup) or an array inclusion (O(N) lookup), e.g. 
```
allowed_content_types = {
'application/json': 1,
....
}
if allowed_content_types.key?(content_type)
  # passed
else 
  # failed
  errors.add....
end
```

Reply:
This is a list of file extensions, which is slightly different from Content-Type eg. `.txt` files have Content-Type `text/plain`. 

I’m not sure if this is validating the file name for an allowed extension or the Content-Type header for an allowed type and sub-type.

Reply:
For us to host the media resources the API expects the mime types to be set on creation of the media resource, e.g. for subjects https://panoptes.docs.apiary.io/#reference/subjects/subject-collection/create-a-subject

those are then validated here in the Medium resource so what we will be validating against is the mime type here stored in the `content_type` attribute on that model. These in turn set the `content-type` header that the blob storage system will return via the HTTP request to fetch the media resource from the CDN domain. 

So we aren't validating file extensions here and why i'm worried about the this simple regex allow invalid mime types to pass.

Reply:
FWIW the server never knows the file extension of the media resource, instead the client tells the API it want's to upload a media resource of content_type X (e.g. `image/jpeg`) and the api stores this media resource in the DB, creates a uniq id for blob storage and returns a writable blob storage URL to the client so it can upload the file to the remote blob storage system. 

At no point does the API see the file that will be uploaded, so we can't inspect the contents of the file etc. This would only be possible post the upload by the client.

Reply:
In PFE, I modified the Panoptes file upload to allow any image, audio or video types plus a limited set of text and application types.
https://github.com/zooniverse/Panoptes-Front-End/pull/6260

Reply:
I could probably have chosen better names for these variables but this is the code that checks Content-Type to see if a file can be uploaded.
https://github.com/zooniverse/Panoptes-Front-End/blob/0a7990d11e3d90600d8043593069ee2704dab3c5/app/lib/put-file.js#L1-L8

Reply:
> Did you consider inverting this check to a block list of non-allowed content-types vs an inclusive allow list?
> Allow lists are harder to maintain as they are more restrictive but can provide finer controls.

I'm fine either way. IMO when I was thinking through this at least, blocklists seemed a bit "harder" since it felt like there would be many things to block (I could be wrong). Basing on this list of types: https://gist.github.com/tylerlee/53609bff1346cebf8f0a85b6be29a88e it felt like there would be too much to block. 

> If we do the allow list I think this regex should match what we are testing, i.e the list in and no more. I would remove the spx, opus, ttf, tar, gz, bz2, tbz2, zip etc. 

Sounds good. I was going by what was mentioned in slack as the regex test, but that apparently was incorrect. 

> Also i would ensure we test the content types in full not substrings (can smuggle by faking content types, e.g. test/html-json will pass).

Ah, yeah thats a good test to add.